pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,173 @@
+    The fucntions, along with scalar coefficients, can then be used to initialize a ParametrizedHamiltonian,
+    which will be split into a fixed and parametrized term. The fixed term is an Operator, while the parametrized
+    term must be initialized with concrete parameters to obtain an Operator.
+    >>> coeffs = [2, f1, f2]
+    >>> obs = [qml.PauliX(0)@qml.PauliX(1), qml.PauliY(0)@qml.PauliY(1), qml.PauliZ(0)@qml.PauliZ(1)]
+    >>> H = ParametrizedHamiltonian(coeffs, obs)
+    >>> H.H_fixed
+      2*(PauliX(wires=[0]) @ PauliX(wires=[1]))
+    >>> H.H_parametrized([2.5, 3.6], t)
+      (0.5984721441039564*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (1.9450883011253033*(PauliZ(wires=[0]) @ PauliZ(wires=[1])))
+
+    The resulting object can be passed parameters, and will return an Operator representing the
+    ParametrizedHamiltonian with the specified parameters:
+
+    >>> H([1.2, 2.3], 0.5)
+    (2*(PauliX(wires=[0]) @ PauliX(wires=[1]))) + ((0.5646424733950354*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (2.0184398923478573*(PauliZ(wires=[0]) @ PauliZ(wires=[1]))))
+
+    """
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+
+    def __init__(
+        self,
+        coeffs,
+        observables,
+    ):
+
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid Hamiltonian; "
+                "number of coefficients and operators does not match."
+            )
+
+        for obs in observables:
+            if not isinstance(obs, Observable):
+                raise ValueError(
+                    "Could not create circuits. Some or all observables are not valid."
+                )
+
+        self._ops = list(observables)
+        self._coeffs = coeffs
+
+        self.H_fixed_coeffs = []
+        self.H_fixed_ops = []
+        self.H_parametrized_fns = []
+        self.H_parametrized_ops = []
+
+        for coeff, obs in zip(coeffs, observables):
+            if callable(coeff):
+                self.H_parametrized_fns.append(coeff)
+                self.H_parametrized_ops.append(obs)
+            else:
+                self.H_fixed_coeffs.append(coeff)
+                self.H_fixed_ops.append(obs)
+
+    def __call__(self, params, t):
+        return self.H_fixed + self.H_parametrized(params, t)
+
+    def __repr__(self):
+        return f"ParametrizedHamiltonian: terms={qml.math.shape(self.coeffs)[0]}"
+
+    @staticmethod
+    def _get_terms(coeffs, obs):
+        """Takes a list of scalar coefficients and list of Observables. Returns a qml.Sum of qml.SProd operators
+        (or a single qml.SProd operator in the event that there is only one term)."""
+        terms_list = [qml.s_prod(coeff, ob) for coeff, ob in zip(coeffs, obs)]
+        if len(terms_list) == 0:
+            return None
+        if len(terms_list) == 1:
+            return terms_list[0]
+        return qml.op_sum(*terms_list)
+
+    @property
```

Comment:
Why is this a property and not a method? Any dis-/advantages of using one over the other?

Reply:
This one is just a property because `H_parametrized` is, see comment below. Even if we keep `H_parametrized` as a property, might make sense to make this a method, so that they are called in the same way to get an operator, i.e.

```
>>> pH.H_fixed()
qml.Sum(op1, op2...)
>>>pH.H_parametrized(params, t)
qml.Sum(op1, op2)
```

instead of current

```
>>> pH.H_fixed
qml.Sum(op1, op2...)
>>>pH.H_parametrized(params, t)
qml.Sum(op1, op2)
```

What are your thoughts (for both this and `H_parametrized`)?

Reply:
Definitely see your point and much prefer this! 💯